### PR TITLE
Added API tokens to marble seeds. Missing UI

### PR DESCRIPTION
**Pull request by Siedrix**
Missing UI on profile(app and admin) page to create, list and revoke API tokens.

### CODE REVIEW

File: api/middlewares/getRequestData.js [hunk 1]
```diff
@@ -29,6 +29,34 @@ module.exports = async function (ctx, next) {
...
+
+      if (!userToken.user) {
+        return ctx.throw(401, 'Invalid User')
+      }
+
+      ctx.state.user = userToken.user
+      ctx.state.token = userToken
+      ctx.state.authMethod = 'Basic'
 
       userToken.lastUse = new Date()
```
Comment (luislizarraga) [inline api/middlewares/getRequestData.js]: These shouldn't be here if the same thing is done in lines 34-35

File: api/routers/user/create-api-token.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+    if (!user) {
+      return ctx.throw(403)
+    }
+
+    if (ctx.state.authMethod !== 'Bearer') {
+      return ctx.throw(403)
+    }
+
+    const token = await await user.createToken({
```
Comment (luislizarraga) [inline api/routers/user/create-api-token.js]: double `await`

File: api/routers/user/delete-api-token.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+
+    if (!currentToken) {
+      return ctx.throw(401)
+    }
+
+    if (ctx.state.authMethod === 'Basic' && currentToken.uuid !== tokenUUId) {
+      return ctx.throw(403)
+    }
+
+    const token = UserToken.findOne({'uuid': tokenUUId})
```
Comment (luislizarraga) [inline api/routers/user/delete-api-token.js]: Missing `await`

File: lib/frontend/components/base-deleteButton/index.js [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+    })
+  }
+
+  hideModal () {
+    this.setState({
+      className: ''
+    })
+  }
+
+  finishUp (object) {
```
Comment (luislizarraga) [inline lib/frontend/components/base-deleteButton/index.js]: We can remove this function, it isn't used

File: lib/frontend/components/tokenList.js [hunk 5]
```diff
@@ -0,0 +1,223 @@
...
+            <div className='message-body is-size-7 has-text-centered'>
+              {error}
+            </div>
+          </div>
+          <div className='field is-grouped'>
+            <div className='control'>
+              <button className='button is-primary'>Create</button>
+            </div>
+            <div className='control'>
+              <button className='button' onClick={this.hideModal}>Cancel</button>
```
Comment (luislizarraga) [inline lib/frontend/components/tokenList.js]: Missing `type` attribute

File: lib/frontend/components/tokenList.js [hunk 6]
```diff
@@ -0,0 +1,223 @@
...
+          </div>
+
+          <div className={this.state.apiCallErrorMessage}>
+            <div className='message-body is-size-7 has-text-centered'>
+              {error}
+            </div>
+          </div>
+          <div className='field is-grouped'>
+            <div className='control'>
+              <button className='button is-primary'>Create</button>
```
Comment (luislizarraga) [inline lib/frontend/components/tokenList.js]: Missing `type` attribute

File: api/routers/user/create-api-token.js (hunk 2 above)
Comment (Siedrix) [reply api/routers/user/create-api-token.js]: You really need to wait... I will fix it

Comment (luislizarraga) [reply api/routers/user/create-api-token.js]: LOL :joy:

**Pull request merged**